examples/generative/vq_vae.py

```diff
@@ -357,7 +366,23 @@ def show_subplot(original, reconstructed):
 print(f"Input shape of the PixelCNN: {pixelcnn_input_shape}")
 
 """
-Don't worry about the input shape. It'll become clear in the following sections.
+This input shape represents the MNIST image size reduction from 28x28 pixels in the original digit
+images, that pass through two layers of stride-2 convolution. With "same" padding, this exactly
+halves the "resolution" of the output shape for each stride-2 layer, so with these two layers we
+end up with an encoder output tensor of 7x7 on axes 2 and 3, with the first axis as the batch size
+and the last axis being the code book embedding size. Since the quantization layer in the
+autoencoder maps these 7x7 tensors to indices of the code book, these output layer axis sizes must
+be matched by the PixelCNN as the input shape. The task of the PixelCNN for this architecture is to
+generate _likely_ 7x7 arrangements of codebook indices.
```

Comment:
```suggestion

This input shape represents the reduction in the resolution performed by the encoder. With "same" padding, this exactly halves the "resolution" of the output shape for each stride-2 convolution layer. So, with these two layers, we
end up with an encoder output tensor of 7x7 on axes 2 and 3, with the first axis as the batch size and the last axis being the code book embedding size. Since the quantization layer in the autoencoder maps these 7x7 tensors to indices of the code book, these output layer axis sizes must be matched by the PixelCNN as the input shape. The task of the PixelCNN for this architecture is to generate _likely_ 7x7 arrangements of codebook indices.
```

Reply:
I accepted your suggestion but this leaves two very long lines in the file, the second one being 444 chars long. Are you sure you intended this @sayakpaul ?

Reply:
Hmm, probably a rendering issue from GitHub's suggestion feature then. If the lines are becoming too long then feel free to break them up.